contents/docs/libraries/next-js/_snippets/app-router-server-setup.mdx

```diff
@@ -0,0 +1,48 @@
+For the app router, we can initialize the `posthog-node` SDK once with a `PostHogClient` function, and import it into files.
+
+This enables us to send events and fetch data from PostHog on the server – without making client-side requests.
+
+```js
+// app/posthog.js
+import { PostHog } from 'posthog-node'
+
+export default function PostHogClient() {
+  const posthogClient = new PostHog(process.env.NEXT_PUBLIC_POSTHOG_KEY, {
+    host: process.env.NEXT_PUBLIC_POSTHOG_HOST,
+    flushAt: 1,
+    flushInterval: 0
+  })
+  return posthogClient
+}
+```
+
+> **Note:** Because server-side functions in Next.js can be short-lived, we set `flushAt` to `1` and `flushInterval` to `0`. 
+> - `flushAt` sets how many capture calls we should flush the queue (in one batch). 
+> - `flushInterval` sets how many milliseconds we should wait before flushing the queue. 
+> Setting them to the lowest number ensures events are sent immediately and not batched. We also need to call `await posthog.shutdown()` once done.
```

Comment:
Do you think it might make sense to recommend using `waitUntil`? https://vercel.com/changelog/waituntil-is-now-available-for-vercel-functions

That way you can do 

```ts
waitUntil(posthog.shutdown())
```

instead of await-ing it, which will slow down rendering.